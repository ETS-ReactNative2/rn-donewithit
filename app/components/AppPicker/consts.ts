import COLORS from '../../config/colors'

const CONSTS = {
  PLACEHOLDER: 'Select options...',
  ICON_NAME: 'email',
  ICON_COLOR: COLORS.GRAY,
  ICON_SIZE: 20,
  BORDER_RADIUS: 25,
  PADDING: 10,
  BACKGROUND_COLOR: COLORS.INPUT_BACKGROUND_COLOR,
  TEXT_INPUT_FONT_SIZE: 18,
  HAS_ICON: true,
  OPTIONS: [],
  OPTION_VALUE: 'value',
  OPTION_LABEL: 'label',
  HEIGHT: 50,
  NUM_COLUMNS: 1,
  HAS_ITEM_SEPARATOR: true
}

export default CONSTS
