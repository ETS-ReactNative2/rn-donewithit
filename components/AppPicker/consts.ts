import COLORS from '../../config/colors'

const CONSTS = {
  PLACEHOLDER: 'Select options...',
  ICON_NAME: 'email',
  ICON_COLOR: COLORS.GRAY,
  ICON_SIZE: 20,
  BORDER_RADIUS: 25,
  PADDING: 10,
  BACKGROUND_COLOR: '#fbf6f8',
  TEXT_INPUT_FONT_SIZE: 18,
  HAS_ICON: true,
  OPTIONS: [],
  OPTION_VALUE: 'value',
  OPTION_LABEL: 'label',
  HEIGHT: 50
}

export default CONSTS
