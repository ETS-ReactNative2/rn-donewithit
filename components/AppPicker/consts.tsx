import COLORS from '../../config/colors'

const CONSTS = {
  PLACEHOLDER: 'Select options...',
  ICON_NAME: 'email',
  ICON_COLOR: COLORS.GRAY,
  ICON_SIZE: 20,
  BORDER_RADIUS: 20,
  PADDING: 10,
  BACKGROUND_COLOR: '#fbf6f8',
  TEXT_INPUT_FONT_SIZE: 18
}

export default CONSTS
