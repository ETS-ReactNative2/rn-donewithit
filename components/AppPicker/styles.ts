import { Platform, StyleSheet } from 'react-native'
import CONSTS from './consts'
import COLORS from '../../config/colors'

const styles = StyleSheet.create({
  container: {
    backgroundColor: CONSTS.BACKGROUND_COLOR,
    borderRadius: CONSTS.BORDER_RADIUS,
    padding: CONSTS.PADDING,
    overflow: 'hidden',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    height: CONSTS.HEIGHT
  },
  icon: {
    marginRight: CONSTS.PADDING
  },
  textInput: {
    ...Platform.select({
      android: {
        fontFamily: 'Roboto'
      },
      ios: {
        fontFamily: 'Courier'
      }
    })
  },
  label: {
    fontSize: CONSTS.TEXT_INPUT_FONT_SIZE,
    flex: 1
  },
  placeholder: {
    color: COLORS.SILVER
  },
  flatListItem: {
    textAlign: 'center',
    lineHeight: 50,
    fontSize: CONSTS.TEXT_INPUT_FONT_SIZE
  },
  button: {
    margin: CONSTS.PADDING
  }
})

export default styles
