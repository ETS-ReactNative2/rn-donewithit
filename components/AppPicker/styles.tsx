import { Platform, StyleSheet } from 'react-native'
import CONSTS from './consts'
import COLORS from '../../config/colors'

const styles = StyleSheet.create({
  container: {
    backgroundColor: CONSTS.BACKGROUND_COLOR,
    borderRadius: CONSTS.BORDER_RADIUS,
    padding: CONSTS.PADDING,
    overflow: 'hidden',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center'
  },
  icon: {
    marginRight: CONSTS.PADDING
  },
  textInput: {
    ...Platform.select({
      android: {
        fontFamily: 'Roboto'
      },
      ios: {
        fontFamily: 'Courier'
      }
    })
  },
  placeholder: {
    fontSize: CONSTS.TEXT_INPUT_FONT_SIZE,
    flex: 1
  }
})

export default styles
