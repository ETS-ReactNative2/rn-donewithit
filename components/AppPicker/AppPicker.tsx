import React, { useCallback } from 'react'
import { View, GestureResponderEvent, TouchableOpacity } from 'react-native'
import AppIcon from '../AppIcon'
import AppText from '../AppText'
import CONSTS from './consts'
import styles from './styles'

function AppPicker({
  style,
  placeholder = CONSTS.PLACEHOLDER,
  onPress,
  // AppIcon
  iconName = CONSTS.ICON_NAME,
  iconColor = CONSTS.ICON_COLOR,
  iconSize = CONSTS.ICON_SIZE
}: any) {
  const handlePress = useCallback((e?: GestureResponderEvent) => {
    typeof onPress === 'function' && onPress(e)
  }, [])
  return (
    <TouchableOpacity activeOpacity={0.68} onPress={handlePress}>
      <View style={[style, styles.container]}>
        <AppIcon
          name={iconName}
          color={iconColor}
          size={iconSize}
          style={styles.icon}
        />
        <AppText style={styles.placeholder}>{placeholder}</AppText>
        <AppIcon name="chevron-down" color={iconColor} size={iconSize} />
      </View>
    </TouchableOpacity>
  )
}

export default AppPicker
