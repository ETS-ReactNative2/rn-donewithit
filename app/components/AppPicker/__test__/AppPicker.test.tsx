import React from 'react'
import renderer from 'react-test-renderer'
import AppPicker from '../index'
import COLORS from '../../../config/colors'

jest.mock(
  '@expo/vector-icons/build/vendor/react-native-vector-icons/lib/create-icon-set.js',
  () => {
    return () => ''
  }
)

const props = {
  icon: false,
  placeholder: 'Category',
  options: [
    {
      label: 'Furniture',
      iconName: 'lamp',
      iconBackgroundColor: COLORS.PRIMARY,
      value: 1
    },
    {
      label: 'Cars',
      iconName: 'car',
      iconBackgroundColor: 'orange',
      value: 2
    },
    {
      label: 'Cameras',
      iconName: 'camera',
      iconBackgroundColor: 'gold',
      value: 3
    },
    {
      label: 'Games',
      iconName: 'cards-playing-outline',
      iconBackgroundColor: 'limegreen',
      value: 4
    },
    {
      label: 'Clothing',
      iconName: 'shoe-heel',
      iconBackgroundColor: 'mediumturquoise',
      value: 5
    },
    {
      label: 'Sports',
      iconName: 'basketball',
      iconBackgroundColor: 'cornflowerblue',
      value: 6
    },
    {
      label: 'Movies & Music',
      iconName: 'headphones',
      iconBackgroundColor: 'royalblue',
      value: 7
    },
    {
      label: 'Books',
      iconName: 'book-open-variant',
      iconBackgroundColor: 'mediumorchid',
      value: 8
    },
    {
      label: 'Other',
      iconName: 'application',
      iconBackgroundColor: 'lightslategrey',
      value: 9
    }
  ],
  numColumns: 3,
  hasItemSeparator: false
}

describe('AppPicker', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<AppPicker {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
