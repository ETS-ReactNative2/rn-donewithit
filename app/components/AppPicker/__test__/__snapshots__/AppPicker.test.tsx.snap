// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppPicker renders correctly 1`] = `
Array [
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        undefined,
        Object {
          "alignItems": "center",
          "backgroundColor": "#f9f2f2",
          "borderRadius": 25,
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "space-between",
          "overflow": "hidden",
          "padding": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "Courier",
            "fontSize": 13,
          },
          Array [
            Object {
              "flex": 1,
              "fontSize": 18,
            },
            Object {
              "color": "#C0C0C0",
            },
          ],
        ]
      }
    >
      Category
    </Text>
    <Text />
  </View>,
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    visible={false}
  >
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      data={
        Array [
          Object {
            "iconBackgroundColor": "tomato",
            "iconName": "lamp",
            "label": "Furniture",
            "value": 1,
          },
          Object {
            "iconBackgroundColor": "orange",
            "iconName": "car",
            "label": "Cars",
            "value": 2,
          },
          Object {
            "iconBackgroundColor": "gold",
            "iconName": "camera",
            "label": "Cameras",
            "value": 3,
          },
          Object {
            "iconBackgroundColor": "limegreen",
            "iconName": "cards-playing-outline",
            "label": "Games",
            "value": 4,
          },
          Object {
            "iconBackgroundColor": "mediumturquoise",
            "iconName": "shoe-heel",
            "label": "Clothing",
            "value": 5,
          },
          Object {
            "iconBackgroundColor": "cornflowerblue",
            "iconName": "basketball",
            "label": "Sports",
            "value": 6,
          },
          Object {
            "iconBackgroundColor": "royalblue",
            "iconName": "headphones",
            "label": "Movies & Music",
            "value": 7,
          },
          Object {
            "iconBackgroundColor": "mediumorchid",
            "iconName": "book-open-variant",
            "label": "Books",
            "value": 8,
          },
          Object {
            "iconBackgroundColor": "lightslategrey",
            "iconName": "application",
            "label": "Other",
            "value": 9,
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "Courier",
                      "fontSize": 13,
                    },
                    Array [
                      Object {
                        "fontSize": 18,
                        "lineHeight": 50,
                        "paddingHorizontal": 20,
                      },
                      false,
                    ],
                  ]
                }
              >
                Furniture
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "Courier",
                      "fontSize": 13,
                    },
                    Array [
                      Object {
                        "fontSize": 18,
                        "lineHeight": 50,
                        "paddingHorizontal": 20,
                      },
                      false,
                    ],
                  ]
                }
              >
                Cars
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "Courier",
                      "fontSize": 13,
                    },
                    Array [
                      Object {
                        "fontSize": 18,
                        "lineHeight": 50,
                        "paddingHorizontal": 20,
                      },
                      false,
                    ],
                  ]
                }
              >
                Cameras
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "Courier",
                      "fontSize": 13,
                    },
                    Array [
                      Object {
                        "fontSize": 18,
                        "lineHeight": 50,
                        "paddingHorizontal": 20,
                      },
                      false,
                    ],
                  ]
                }
              >
                Games
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "Courier",
                      "fontSize": 13,
                    },
                    Array [
                      Object {
                        "fontSize": 18,
                        "lineHeight": 50,
                        "paddingHorizontal": 20,
                      },
                      false,
                    ],
                  ]
                }
              >
                Clothing
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "Courier",
                      "fontSize": 13,
                    },
                    Array [
                      Object {
                        "fontSize": 18,
                        "lineHeight": 50,
                        "paddingHorizontal": 20,
                      },
                      false,
                    ],
                  ]
                }
              >
                Sports
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "Courier",
                      "fontSize": 13,
                    },
                    Array [
                      Object {
                        "fontSize": 18,
                        "lineHeight": 50,
                        "paddingHorizontal": 20,
                      },
                      false,
                    ],
                  ]
                }
              >
                Movies & Music
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "Courier",
                      "fontSize": 13,
                    },
                    Array [
                      Object {
                        "fontSize": 18,
                        "lineHeight": 50,
                        "paddingHorizontal": 20,
                      },
                      false,
                    ],
                  ]
                }
              >
                Books
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "Courier",
                      "fontSize": 13,
                    },
                    Array [
                      Object {
                        "fontSize": 18,
                        "lineHeight": 50,
                        "paddingHorizontal": 20,
                      },
                      false,
                    ],
                  ]
                }
              >
                Other
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Array [
          Object {},
          Object {
            "margin": 10,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "tomato",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "Courier",
                "fontSize": 13,
              },
              Object {
                "color": "#fff",
                "fontSize": 18,
                "fontWeight": "bold",
                "textTransform": "uppercase",
              },
            ]
          }
        >
          close
        </Text>
      </View>
    </View>
  </Modal>,
]
`;
